### ISSUE TITLE
Using relative paths in pybids layout so the container can use them

### ISSUE THREAD

**Issue by nikhil153** [neurodatascience/nipoppy#160]
@Remi-Gau / @surchs (bids in browser) 
I am trying to generate BIDSLayout using Remi's [suggestion](https://bids-standard.github.io/pybids/layout/index.html#loading-bids-datasets) prior to running the mriqc / fmriprep pipelines . 
The newer pybids suggests for that  `BIDSLayoutIndexer` so I am using it. 

```
indexer = BIDSLayoutIndexer(ignore=ignore_list)
layout = BIDSLayout(bids_dir, indexer=indexer, absolute_paths=False)
```
However, the indexed bids_db seem to have absolute paths regardless of the `absolute_paths=False`

```
ValueError: '/home/nikhil/projects/Parkinsons/nimhans/data/PD_YLO/bids/sub-YLOPD03/ses-01/anat' is not in the subpath of '/data' OR one path is relative and the other is absolute.
```

Any ideas?

**Comment by Remi-Gau** [comment]
created the following test on the HEAD of the pybids master branch

```python
def test_relative_paths():
    data_dir = join(get_test_data_path(), 'ds005')
    indexer = BIDSLayoutIndexer()
    layout_rel = BIDSLayout(data_dir, indexer=indexer, absolute_paths=True)
    data_rel = layout_rel.get(subject='01', suffix='bold', run='01')
    print(data_rel)
    layout_abs = BIDSLayout(data_dir, indexer=indexer, absolute_paths=False)
    data_abs = layout_abs.get(subject='01', suffix='bold', run='01')
    print(data_abs)
```

It give me this:

```
[<BIDSImageFile filename='/home/remi/github/bids/pybids/bids/tests/data/ds005/sub-01/func/sub-01_task-mixedgamblestask_run-01_bold.nii.gz'>]

[<BIDSImageFile filename='sub-01/func/sub-01_task-mixedgamblestask_run-01_bold.nii.gz'>]
```

But also the following warning, that says it is better to not use `absolute_paths`...

```
bids/layout/tests/test_layout.py::test_relative_paths
  /home/remi/github/bids/pybids/bids/layout/validation.py:39: UserWarning: The absolute_paths argument will be removed from PyBIDS in 0.14. You can easily access the relative path of BIDSFile objects via the .relpath attribute (instead of .path). Switching to this pattern is strongly encouraged, as the current implementation of relative path handling is known to produce query failures in certain edge cases.
    warnings.warn("The absolute_paths argument will be removed from PyBIDS "
```

**Comment by Remi-Gau** [comment]
suggestion: do the indexing from within the fmriprep container ?

**Comment by nikhil153** [comment]
Yes, I get the same warning about absolute paths as well. So I am hesitant to use it. 
I couldn't find a "non-deprecating-soon" solution - do you know what they propose going forward? 

>  do the indexing from within the fmriprep container ?
That defeats the purpose of doing it automatically before `mriqc` and `fmriprep`. Which helps avoid issues with batch / parallel processing. Also won't that require modifying the container with every version update?

**Comment by nikhil153** [comment]
Fixed by #163
